### PR TITLE
Update to libminesweeper v1.0.0

### PR DESCRIPTION
**Pull request by joelekstrom**
New libminesweeper API that looks a lot better.

Will also distribute mines in `minesweeper_init` instead of when opening the first tile. This should remove the slight lag because of random number generation.

### CODE REVIEW

File: game.c [hunk 1]
```diff
@@ -503,55 +504,47 @@ void play_game(int8_t difficulty)
...
-			move_cursor(board, LEFT, true);
+			minesweeper_move_cursor(game, LEFT, true);
 		if (button_pressed(J_RIGHT, 0))
-			move_cursor(board, RIGHT, true);
+			minesweeper_move_cursor(game, RIGHT, true);
 		if (button_pressed(J_A, -1))
 		{
-			// Add more entropy to prng first time we open a tile
-			if (first_open)
+			if (game->state == MINESWEEPER_PENDING_START)
```
Comment (andreasjhkarlsson) [inline game.c:515]: Please re-add the comment.

File: game.c [hunk 2]
```diff
@@ -503,55 +504,47 @@ void play_game(int8_t difficulty)
...
 			{
-				// 2nd seed
-				seed_prng();
```
Comment (andreasjhkarlsson) [inline game.c:528]: This should not be removed. Without it, only 256 unique games are possible.

File: game.c [hunk 3]
```diff
@@ -439,14 +439,15 @@ void stop_timer()
 	enable_interrupts();
 }
 
-void render_tile(struct board* board, uint8_t *tile, int x, int y)
+void render_tile(struct minesweeper_game* game, uint8_t *tile)
 {
 	uint8_t sprite;
+	unsigned x, y; minesweeper_get_tile_location(game, tile, &x, &y);
```
Comment (andreasjhkarlsson) [inline game.c:438]: Hmm, okejdå.

**Review by andreasjhkarlsson — CHANGES_REQUESTED** [review]
Some things need fixing.

**Comment by joelekstrom** [comment]
The PRNG-concerns should now be fixed

**Comment by andreasjhkarlsson** [comment]
This doesn't work as [button_pressed doesn't return until the button has been released](https://github.com/rotmoset/gb-mines/blob/master/utils.c#L16). Also, using a flag here is not needed. A simple while loop would be simpler, like so:

```
if (joypad() & J_START)
{
   seed_prng();
   while(joypad()&J_START); // Block until release
   seed_prng();
  // ...
}
```

**Comment by andreasjhkarlsson** [comment]
Note: Passing '0' as the second parameter to button_pressed makes it return immedietely without waiting for release.

**Comment by joelekstrom** [comment]
Concerns fixed.

**Pull request merged**